src/visualization/types/Map/view.tsx

```diff
@@ -21,51 +27,97 @@ const GeoPlot: FC<Props> = ({result, properties}) => {
     detectCoordinateFields,
     mapStyle,
   } = properties
-  const tileServerConfiguration = {
-    tileServerUrl: '',
-    bingKey: '',
-  }
 
   const {lat, lon} = properties.center
 
-  let calculatedGeoCoordinates = {
+  const [errorCode, setErrorCode] = useState<string>('')
+  const [mapToken, setMapToken] = useState<string>('')
+  const [geoCoordinates, setGeoCoordinates] = useState<Coordinates>({
     lat,
     lon,
-  }
-  try {
-    calculatedGeoCoordinates = getGeoCoordinates(result.table, 0)
-  } catch (err) {
-    const error =
-      'Map type is not supported with the data provided: Missing latitude/longitude values'
+  })
+
+  useEffect(() => {
+    const getToken = async () => {
+      try {
+        const {token} = await getMapToken()
+        setMapToken(token)
+      } catch (err) {
+        setErrorCode('MapServiceUnavailable')
```

Comment:
can you pull `'MapServiceUnavailable'` and `'InvalidCoordinates'` out into a constant or a TypeScript enum?

just confirming: we can only have one error state, right?

Reply:
We have two error state:
1. when data provided doesn't include lat and lon
2. when map service comes back with an error (unauthorized or forbidden)